src/ListGroup.js

```diff
@@ -13,17 +13,15 @@ class ListGroup extends React.Component {
 
     if (!this.props.children) {
       return this.renderDiv(items);
-    } else if (React.Children.count(this.props.children) === 1 && !Array.isArray(this.props.children)) {
-      let child = this.props.children;
-
-      childrenAnchors = this.isAnchor(child.props);
-
     } else {
+      React.Children.forEach(this.props.children, (child) => {
+        if (!Array.isArray(child) && this.isAnchor(child.props)) {
+          childrenAnchors = true;
+        }
 
-      childrenAnchors = Array.prototype.some.call(this.props.children, (child) => {
-        return !Array.isArray(child) ? this.isAnchor(child.props) : Array.prototype.some.call(child, (subChild) => {
-            return this.isAnchor(subChild.props);
-        });
+        if (Array.isArray(child) && child.some.call((subChild) => this.isAnchor(subChild.props))) {
```

Comment:
the original code does seem to test two layers down, why just two? @cassus that probably means your `forEach` code needs to be pulled out into a separate function so it can be called recursively